Libplanet/Net/Swarm.cs

```diff
@@ -1686,61 +1702,30 @@ private bool IsDemandNeeded(BlockHeader target)
             CancellationToken cancellationToken
         )
         {
-            int retry = 3;
             long previousTipIndex = blockChain.Tip?.Index ?? -1;
             BlockChain<T> synced = null;
-            StateCompleterSet<T> trustedStateCompleterSet = await GetTrustedStateCompleterAsync(
-                trustedStateValidators,
-                dialTimeout,
-                cancellationToken: cancellationToken
-            );
 
             try
             {
-                while (true)
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    try
-                    {
-                        long currentTipIndex = blockChain.Tip?.Index ?? -1;
-                        long receivedBlockCount = currentTipIndex - previousTipIndex;
+                long currentTipIndex = blockChain.Tip?.Index ?? -1;
+                long receivedBlockCount = currentTipIndex - previousTipIndex;
 
-                        FillBlocksAsyncStarted.Set();
-                        synced = await FillBlocksAsync(
-                            peer,
-                            blockChain,
-                            stop,
-                            progress,
-                            totalBlockCount,
-                            receivedBlockCount,
-                            true,
-                            cancellationToken
-                        );
-                        break;
-                    }
-                    catch (TimeoutException e)
-                    {
-                        if (retry > 0)
-                        {
-                            _logger.Error(
-                                e,
-                                $"{nameof(FillBlocksAsync)}() failed; retrying...: {e}"
-                            );
-                            retry--;
-                        }
-                        else
-                        {
-                            FillBlocksAsyncFailed.Set();
-                            throw;
-                        }
-                    }
-                    catch (Exception)
-                    {
-                        FillBlocksAsyncFailed.Set();
-                        throw;
-                    }
-                }
+                FillBlocksAsyncStarted.Set();
+                synced = await FillBlocksAsync(
+                    peer,
+                    blockChain,
+                    stop,
+                    progress,
+                    totalBlockCount,
+                    receivedBlockCount,
+                    true,
+                    cancellationToken
+                );
+            }
+            catch (Exception)
+            {
+                FillBlocksAsyncFailed.Set();
+                throw;
```

Comment:
Shouldn't this be recorded as a behavioral change?

Reply:
Is it okay to port commits treated as behavioral change not bugfix, into 0.10-maintenance not main?

Reply:
IMHO it's more appropriate to be ported to the main rather than 0.10-maintenance.